core/src/main/kotlin/com/malinskiy/marathon/Marathon.kt

```diff
@@ -160,13 +168,14 @@ class Marathon(val configuration: Configuration) {
         deviceProvider.terminate()
     }
 
-    private fun printSummary(scheduler: Scheduler, executionTime: Long) {
-        val pools = scheduler.getPools()
-        if (!pools.isEmpty()) {
-            val summaryPrinter = loadSummaryPrinter()
-            val summary = summaryCompiler.compile(scheduler.getPools())
-            printCliReport(summary, executionTime)
-            summaryPrinter.print(summary)
+    private fun printSummary(scheduler: Scheduler, executionTime: Long, dummyReportPool: DevicePoolId) {
+        scheduler.getPools().also {
```

Comment:
if you are using also, why not to use scheduler.getPools()?.takeIf { it.isNotEmpty() }?.also/run/let